tests/Linter.test.cpp

```diff
@@ -1662,7 +1662,7 @@ TEST_CASE_FIXTURE(Fixture, "WrongCommentOptimize")
 {
     LintResult result = lint(R"(
 --!optimize
---!optimize   
+--!optimize
```

Comment:
The white space here was essential.

Reply:
Should probably extract this into a separate string so the line ends with " instead of whitespace, otherwise editors may trim this. Or don't use `R"()"` for this case.

Reply:
My editor is set up to trim trailing whitespace by default, which isn't uncommon. I think I will make this not a raw string.

Reply:
Yup, that works, just making sure we don't lose this - the trailing whitespace was specifically testing extra handling of whitespace skipping for linting these directives.